### PR TITLE
[Merged by Bors] - fix: rename `LocallyFiniteOrderTop.ofIic` to `LocallyFiniteOrderBot`

### PR DESCRIPTION
**Pull request by bustercopley**
It's about `LocallyFiniteOrderBot`, so the current name is simply wrong.

Also fix documentation mistakes. These errors were already present in Mathlib 3, before the port.

---

It appears that these are simply minor slips that were not noticed during review.
* Introduced in [mathlib3#14430](https://github.com/leanprover-community/mathlib/pull/14430)
* Ported in #1754

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by grunweg — APPROVED** [review]
Looks good to me; thank you for doing this!

File: Mathlib/Order/LocallyFinite.lean [hunk 1]
```diff
@@ -230,16 +230,17 @@ def LocallyFiniteOrderBot.ofIic' (α : Type*) [Preorder α] [DecidableRel ((· 
...
-def LocallyFiniteOrderTop.ofIic (α : Type*) [PartialOrder α] [DecidableEq α]
+def LocallyFiniteOrderBot.ofIic (α : Type*) [PartialOrder α] [DecidableEq α]
     (finsetIic : α → Finset α) (mem_Iic : ∀ a x, x ∈ finsetIic a ↔ x ≤ a) :
     LocallyFiniteOrderBot α :=
   { finsetIic
     finsetIio := fun a => (finsetIic a).filter fun x => x ≠ a
     finset_mem_Iic := mem_Iic
     finset_mem_Iio := fun a x => by rw [mem_filter, mem_Iic, lt_iff_le_and_ne] }
-#align locally_finite_order_top.of_Iic LocallyFiniteOrderTop.ofIic
+-- Note: this is in the wrong namespace in Mathlib 3.
```
Comment (grunweg) [inline Mathlib/Order/LocallyFinite.lean:242]:
```suggestion
-- Note: this was in the wrong namespace in Mathlib 3.

File: Mathlib/Data/Pi/Interval.lean [hunk 2]
```diff
@@ -61,7 +61,7 @@ section LocallyFiniteOrderBot
 variable [∀ i, LocallyFiniteOrderBot (α i)] (b : ∀ i, α i)
 
 instance instLocallyFiniteOrderBot : LocallyFiniteOrderBot (∀ i, α i) :=
-  LocallyFiniteOrderTop.ofIic _ (fun b => piFinset fun i => Iic (b i)) fun b x => by
+  LocallyFiniteOrderBot.ofIic _ (fun b => piFinset fun i => Iic (b i)) fun b x => by
```
Comment (YaelDillies) [inline Mathlib/Data/Pi/Interval.lean:64]: Here's a good way to make sure the name stays correct:
```suggestion
  .ofIic _ (fun b => piFinset fun i => Iic (b i)) fun b x => by
```

**Comment by YaelDillies** [comment]
maintainer merge

**Comment by ocfnash** [comment]
bors merge

**Pull request closed without merging**